internal/devices/update.go

```diff
@@ -51,48 +53,53 @@ func (c *Client) Update() *cobra.Command {
 
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cmd.SilenceUsage = true
-			req := &packngo.DeviceUpdateRequest{}
+			deviceUpdate := metal.NewDeviceUpdateInput()
 
 			if hostname != "" {
-				req.Hostname = &hostname
+				deviceUpdate.Hostname = &hostname
 			}
 
 			if description != "" {
-				req.Description = &description
+				deviceUpdate.Description = &description
 			}
 
 			if userdata != "" {
-				req.UserData = &userdata
+				deviceUpdate.Userdata = &userdata
 			}
 
 			if locked {
-				req.Locked = &locked
+				deviceUpdate.Locked = &locked
 			}
 
 			if len(tags) > 0 {
-				req.Tags = &tags
+				deviceUpdate.Tags = tags
 			}
 
 			if alwaysPXE {
-				req.AlwaysPXE = &alwaysPXE
+				deviceUpdate.AlwaysPxe = &alwaysPXE
```

Comment:
Let's address https://github.com/equinix/metal-cli/issues/257 while we are here.

Reply:
plan to address in a seperate PR. closing the thread.